### ISSUE TITLE
Default behavior of tf.layers.batch_normalization

### ISSUE THREAD

**Issue by hsgkim** [tensorflow/tensorflow#21229]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04.5 LTS (Xenial Xerus)
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: pip (binary)
- **TensorFlow version (use command below)**: (tf.GIT_VERSION, tf.VERSION) == ('v1.9.0-0-g25c197e023', '1.9.0')
- **Python version**: 3.5.2
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: 9.0/6.1
- **GPU model and memory**: NVIDIA GeForce GTX 1080Ti 11177MiB
- **Exact command to reproduce**: tf.layers.batch_normalization
### Describe the problem
I was trying to train my network using multiple batch normalization layers(`tf.layers.batch_normalization`), only to fail the training process even though I was basically copying a published paper. I had skimmed through the documentation as it looked fairly straightforward, especially after already having had added multiple layers using `tf.layers`. Only after a couple days (as training takes quite a long time) did I realize I had missed the note part of the documentation. This is of course my fault for not reading the documentation thoroughly, but it really strikes me odd that the default behavior of this layer is not to update the ops right away. I guess it might be due to performance reasons, but if that is the case why not just have `defer_updates` argument which defaults to False? I think most beginners would expect the layer to update everything right away, and maybe not even expect everything to be fast anyway. Only power users would really want to fine-tune and optimize their network, in which case then they can set `defer_updates` to True.

I think I just rambled a lot here; my point is that I think it would be better if `tf.layers.batch_normalization` would update ops without the need of `update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)` and `with tf.control_dependencies(update_ops)`, and for people who do not want this, throw `defer_updates=True`.

Edit: formatting

**Comment by ppwwyyxx** [comment]
The same feature request was made in #14809, #14699, but got no response.

The feature was supported in several other modules but not tf.layers. e.g.:
[tf.contrib.layers.batch_norm](https://www.tensorflow.org/api_docs/python/tf/contrib/layers/batch_norm): set `updates_collections` to None.
[tensorpack.models.BatchNorm](http://tensorpack.readthedocs.io/modules/models.html#tensorpack.models.BatchNorm): set `internal_update` to True.

**Comment by hsgkim** [comment]
How come the contrib version has `update_collections`, but the one under `tf.layers` does not have it?

I think the Note section in the documentation of [tf.layers.batch_normalization](https://www.tensorflow.org/api_docs/python/tf/layers/batch_normalization) should at least be promoted to a Warning. So far when I followed all the official tutorials, it was okay to just skim through or ignore the Note sections but this really is not the case here.

**Comment by cy89** [comment]
@nealwu it looks like you last updated the documentation on this complexity. Can you please comment on the Note->Warning request from @hsgkim, and either address or reassign to someone who owns this interface?

**Comment by nealwu** [comment]
@hsgkim What did you mean by promoting to a Warning? The "Note:" provides special formatting to the text and displays it prominently in its own box, but I'm not aware of a Warning tag that does this.

**Comment by hsgkim** [comment]
@nealwu Sorry, I meant a Caution like in [this page](https://www.tensorflow.org/guide/saved_model).

**Comment by hsgkim** [comment]
I made a PR regarding this too if you guys think it's appropriate to change the Note to something more serious (a Caution in this case). Thanks!

**Comment by cy89** [comment]
@nealwu up to you.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @nealwu, @cy89: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @nealwu, @cy89: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @nealwu, @cy89: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @nealwu, @cy89: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @nealwu, @cy89: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @nealwu, @cy89: It has been 89 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @nealwu, @cy89: It has been 104 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by nealwu** [comment]
Closing as this method is now deprecated in favor of the Keras method which seems to no longer have this issue, and the PR is missing/closed. I believe the Caution tag is also intended more for things like the security risks mentioned above in any case. Thanks for bringing this up though @hsgkim.
